lib/Dialect/AIEVec/Transforms/VectorToAIEVecConversions.cpp

```diff
@@ -2396,9 +2396,30 @@ struct ComputeNegOpPattern : public OpConversionPattern<arith::NegFOp> {
   }
 };
 
+// Check whether the value of constant operation is int type and the dense value
+// is -1.
+static bool hasConstNegOneValue(arith::ConstantOp constOp, unsigned elWidth) {
+  if (!constOp) {
+    return false;
+  }
+  auto cstDense = dyn_cast<DenseIntElementsAttr>(constOp.getValue());
+  if (!cstDense) {
+    return false;
+  }
+
+  if (elWidth == 32) {
+    return cstDense.getSplatValue<int32_t>() == -1;
+  } else if (elWidth == 16) {
+    return cstDense.getSplatValue<int16_t>() == -1;
+  } else {
```

Comment:
In case that elWidth == 64 the function would return the incorrect result if any of the top 56 bits is 0.